src/deploy/util.js

```diff
@@ -111,6 +111,7 @@ const waitForDeploy = async (api, deployId, siteId, timeout) => {
 const getUploadList = (required, shaMap) => {
   if (!required || !shaMap) return []
   // TODO: use `Array.flatMap()` instead once we remove support for Node <11.0.0
+  // eslint-disable-next-line unicorn/prefer-spread
   return [].concat(...required.map((sha) => shaMap[sha]))
```

Comment:
@ehmicky what do think about [this change](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1054). I don't think array spread can be easily used to flatten arrays like `concat`.
However on Node.js >=11 `Array.flat` would be a better choice.

Reply:
The suggested change only works when the number of array to flattened is known (as in the examples in the PR). This is not the case for `required` above.

Reply:
It looks like `required` is an array of strings, not an array of arrays though.

https://github.com/netlify/js-client/blob/8b01b76c2caf4772b3445b45cc28c67d51e303f5/src/deploy/index.js#L104

https://github.com/netlify/open-api/blob/master/swagger.yml#L2321

If that's the case, then the following should work:

```js
  return required.map((sha) => shaMap[sha])
```

Reply:
I believe `shaMap` is a mapping between a string (`sha`) to an array of objects.
We would like to get a flattened array of all those objects.
Please correct me if I'm wrong.

Reply:
Oh, right!
Then we probably need to keep the current code as is.